frontend/src/components/organisms/EventForm.tsx

```diff
@@ -80,15 +335,31 @@ const EventForm = ({ eventType }: EventFormProps) => {
       </div>
       <div className="sm:space-x-4 grid grid-cols-1 sm:grid-cols-2">
         <div className="pb-4 sm:pb-0">
-          <DatePicker label="Start Date" />
+          <DatePicker
+            label="Start Date"
+            value={eventDetails ? eventDetails.startDate : undefined}
+            onChange={(e) => (e.$d != "Invalid Date" ? setStartDate(e.$d) : "")}
+          />
         </div>
-        <DatePicker label="End Date" />
+        <DatePicker
+          label="End Date"
+          value={eventDetails ? eventDetails.endDate : undefined}
+          onChange={(e) => (e.$d != "Invalid Date" ? setEndDate(e.$d) : "")}
+        />
       </div>
       <div className="sm:space-x-4 grid grid-cols-1 sm:grid-cols-2">
         <div className="pb-4 sm:pb-0">
-          <TimePicker label="Start Time" />
+          <TimePicker
+            label="Start Time"
+            value={eventDetails ? eventDetails.startTime : undefined}
+            onChange={(e) => (e.$d != "Invalid Date" ? setStartTime(e.$d) : "")}
+          />
         </div>
-        <TimePicker label="End Time" />
+        <TimePicker
+          label="End Time"
+          value={eventDetails ? eventDetails.endTime : undefined}
+          onChange={(e) => (e.$d != "Invalid Date" ? setEndTime(e.$d) : "")}
```

Comment:
it appears `e` could be null which will lead to errors like this
<img width="1065" alt="Screenshot 2023-11-20 at 16 38 54" src="https://github.com/cornellh4i/lagos-volunteers/assets/59776300/df591f39-5f16-4f61-b9e8-5e5b81d87da8">

I think you need something like notice the question mark before `$d`
`onChange={(e) => (e?.$d != "Invalid Date" ? setEndTime(e.$d) : "")}`